### ISSUE TITLE
Package resolution become extraneous.

### ISSUE THREAD

**Issue by janzenz** [rogeriochaves/npm-force-resolutions#8]
Transitive dependency seem to end up `extraneous`. For example I added:
```
  "resolutions": {
    "tar": "~2.2.2"
  }
```
I get `npm ERR! extraneous: tar@2.2.2`.

My specs:
```
node@8.9.4
```

Is this expected?

**Comment by kuc** [comment]
Seems that you can't use `^` also

**Comment by marcodisarno** [comment]
Just a word of defense here - why would you want to use either of `~` and `^` with resolutions?

The whole point here is to exactly not do that, because that's usually how the unexpected bug/security leak came in in the first place.

Imho you have to decide: trust the downstream, or cherry-pick so you can rely on your tests. Second of which is what resolutions is for.

**Comment by kuc** [comment]
@marcodisarno  It's a reasonable behavior, but just not documented clearly enough.

**Comment by henry-stroud** [comment]
did you find a resolution for this?

**Comment by janzenz** [comment]
@henry-stroud I don't think so. I've already moved from the project that used this. So I left it as is, sadly.
